pkg/runtime/wfengine/wfengine_test.go

```diff
@@ -714,6 +715,60 @@ func TestRaiseEvent(t *testing.T) {
 	}
 }
 
+// TestPurge verifies that a workflow can have a series of activites created and then
+// verifies that all the metadata for those activities can be deleted from the statestore
+func TestPurge(t *testing.T) {
+	r := task.NewTaskRegistry()
+	r.AddOrchestratorN("ActivityChainToPurge", func(ctx *task.OrchestrationContext) (any, error) {
+		val := 0
+		for i := 0; i < 10; i++ {
+			if err := ctx.CallActivity("PlusOne", val).Await(&val); err != nil {
+				return nil, err
+			}
+		}
+		return val, nil
+	})
+	r.AddActivityN("PlusOne", func(ctx task.ActivityContext) (any, error) {
+		var input int
+		if err := ctx.GetInput(&input); err != nil {
+			return nil, err
+		}
+		return input + 1, nil
+	})
+
+	ctx := context.Background()
+	client, engine, stateStore := startEngineAndGetStore(ctx, r)
+	for _, opt := range GetTestOptions() {
+		t.Run(opt(engine), func(t *testing.T) {
+			id, err := client.ScheduleNewOrchestration(ctx, "ActivityChainToPurge")
+			if assert.NoError(t, err) {
+				metadata, err := client.WaitForOrchestrationStart(ctx, id)
+				if assert.NoError(t, err) {
+					assert.Equal(t, id, metadata.InstanceID)
+					metadata, err = client.FetchOrchestrationMetadata(ctx, id)
+					time.Sleep(5 * time.Second) // The sleep is here to avoid purging an on-going activity
+
+					err := client.PurgeOrchestrationState(ctx, id)
+					assert.NoError(t, err)
+
+					// Check that no key from the statestore that was expected to be purged is still present in the statestore
+					keysPostPurge := []string{}
+					for key := range stateStore.(*fakeStateStore).items {
+						keysPostPurge = append(keysPostPurge, key)
+					}
+
+					for _, item := range keysPostPurge {
+						// Note that "activityState" comes from "getActivityInvocationKey" inside activity.go
+						if strings.Contains(item, "history") || strings.Contains(item, "activityState") || strings.Contains(item, "inbox") {
```

Comment:
Alternatively, can we make sure there aren't any keys that contain the workflow instance ID? That might be less likely to result in false positives. Otherwise, if we add new fields to the stored workflow state we have to remember to also update this code, which would be really easy to forget.